models/reviewDao.js

```diff
@@ -0,0 +1,125 @@
+const appDataSource = require('./appDataSource');
+const { CustomError } = require('../utils/error');
+
+const getReviewsByMusicalId = async (musicalId, limit, offset) => {
+  try {
+    return await appDataSource.query(
+      `SELECT 
+  reviews.id reviewId,
+  users.profile_nickname as profileNickname,
+  reviews.content,
+  reviews.score,
+  images.reviewImages as reviewImages,
+  reviews.created_at as createTime
+  FROM reviews
+  LEFT JOIN users ON reviews.user_id  = users.id 
+  LEFT JOIN (
+  SELECT 
+  review_images.review_id,
+  JSON_ARRAYAGG( review_images.review_images_url) as reviewImages
+  FROM review_images 
+  GROUP BY review_images.review_id 
+  ) as images ON images.review_id = reviews.id 
+  WHERE reviews.musical_id  = ?
+  ORDER BY reviews.created_at  DESC
+  LIMIT ? OFFSET ?`,
+      [musicalId, limit, offset]
+    );
+  } catch (err) {
+    throw new CustomError(400, 'dataSource_Error');
+  }
+};
+
+const getAverageReviewScore = async (musicalId) => {
+  try {
+    let [result] = await appDataSource.query(
+      `	SELECT
+      reviews.musical_id as musicalId,
+      ROUND(COALESCE(AVG(reviews.score),0),1)  as averageScore
+      FROM reviews 
+      WHERE reviews.musical_id = ?
+      GROUP BY reviews.musical_id`,
+      [musicalId]
+    );
+    if (!result) {
+      result = { averageScore: 0 };
+    }
+    return result;
+  } catch (err) {
+    throw new CustomError(400, 'dataSource_Error');
+  }
+};
```

Comment:
- 아래와 같이 삼항 연산자를 사용하여서 조금더 가독성을 확보할 수 있곘네요!

```suggestion
const getAverageReviewScore = async (musicalId) => {
  try {
    const [result] = await appDataSource.query(
      `SELECT
      reviews.musical_id as musicalId,
      ROUND(COALESCE(AVG(reviews.score),0),1)  as averageScore
      FROM reviews 
      WHERE reviews.musical_id = ?
      GROUP BY reviews.musical_id`,
      [musicalId]
    );
    return { averageScore: result ? result.averageScore : 0 };
  } catch (err) {
    throw new CustomError(400, 'dataSource_Error');
  }
};
```